File: src/models/control_vec.py
"""
control_vec - Função para gerar o vetor de controle da aeronave
Descrição: Constrói o vetor de controle a partir do vetor de incógnitas.
Entrada:
- x (numpy.ndarray): Vetor de incógnitas (15 elementos).
Saída:
- U (numpy.ndarray): Vetor de controle (6 elementos).
"""

import numpy as np

def control_vec(x):
    """
    Gera o vetor de controle da aeronave.

    Parâmetros:
    - x: numpy.ndarray, vetor de incógnitas (15 elementos).

    Retorno:
    - U: numpy.ndarray, vetor de controle (6 elementos).
    """
    # Inicialização do vetor de controle
    U = np.zeros(6)

    # Controles diretamente extraídos de 'x'
    U[0] = x[9]  # Empuxo da esquerda (Tle)
    U[1] = x[10]  # Empuxo da direita (Tre)
    U[2] = x[11]  # Ângulo do estabilizador horizontal (ih)
    U[3] = x[12]  # Deflexão do profundor (de)
    U[4] = x[13]  # Deflexão dos ailerons (da)
    U[5] = x[14]  # Deflexão do leme de direção (dr)

    return U

File: docs/AB/Exame_final/2/control_vec.m
function U = control_vec(x)

% X = [1   2   3    4   5 6  7    8   9  10  11   12 | 13  14  15 16 17 18]
% X = [V alpha q  theta H x beta phi  p  r   psi  y  | Tle Tre ih de da dr]
% x = [1   2   3    4   5 -  6    7   8  9  -    -   | 10  11  12 13 14 15]
U = zeros(6,1);
U(1) = x(9);
U(2) = x(10);
U(3) = x(11);
U(5) = x(12);
U(6) = x(13);
